src/app/admin/group/group.service.ts

```diff
@@ -0,0 +1,43 @@
+import { Injectable } from '@angular/core';
+import { Group, Speciality, Faculty } from '../../shared/entity.interface';
+import { ApiService } from 'src/app/shared/services/api.service';
+import { ModalService } from '../../shared/services/modal.service';
+
+@Injectable({
+  providedIn: 'root'
+})
+export class GroupService {
+  listGroups: Group[] = [];
+  listSpeciality: Speciality[] = [];
+  listFaculty: Faculty[] = [];
+  constructor(private apiService: ApiService, private modalService: ModalService) { }
+
+  getListSpeciality() {
+    this.apiService
```

Comment:
I think it would be better to return data which is wrapped in observable.
return this.isListGroupsLoaded ? of(this.listSpeciality) : this.apiService.getEntity('Speciality').pipe(tap(() => {this.isListGroupsLoaded = true}))